drivers/Kconfig

```diff
@@ -88,18 +88,6 @@ if CAN
 source drivers/can/Kconfig
 endif
 
-config ARCH_HAVE_PWM_PULSECOUNT
-	bool
-	default n
-
-config ARCH_HAVE_PWM_MULTICHAN
-	bool
-	default n
-
-config ARCH_HAVE_I2CRESET
-	bool
-	default n
-
```

Comment:
I believe that it is an error to move ARCH_HAVE_I2CRESET so that it depencds on I2C.  There is no dependency on I2C at all.  Setting that option simply means that the architecture does dupport I2CRESET.  I2C is not needed.

Further, this introduces errors. Abdelatif points out this the causes warnings in the Kconfig files.  Since there is absolutely no logical dependency on I2C, architecture configurations select ARCH_HAVE_I2CRESERT (correct) but because of this change, all of this selections now cause un-met dependency warnings incorrect.

The correct fix is to restore the code as it was before (and NOT to add if I2C).

Reply:
I think this applies in general to all settings of the form ARCH_HAVE_*.  They should never be conditioned on anything.

These are simple announcements that the architecture supports a feature and must be unconditional.

Reply:
@patacongo I will provide a patch move all HAVE to original place.